### PR TITLE
Implement :set command

### PR DESCRIPTION
**Pull request by yuzibo**
See #1178 
Now, i just add it into the frame of the icommands.
Should i push it now?
Signed-off-by: Bo YU <yuzibode@126.com>

* **What does this PR do?**

* **Screenshots (if relevant)**

* **Does this PR meet the acceptance criteria?** (This is just a reminder for you,
  this section can be removed if you fulfill it.)

   - Documentation created/updated (you have to edit
     [doc/manual.xml.head](https://www.github.com/neomutt/neomutt/blob/master/doc/manual.xml.head)
     for that)

   - All builds and tests are passing

   - Added [doxygen code documentation](https://neomutt.org/dev/doxygen)
     [syntax](http://www.stack.nl/~dimitri/doxygen/manual/docblocks.html)

   - Code follows the [style guide](https://neomutt.org/dev/coding-style)

* **What are the relevant issue numbers?**
#1178

### CODE REVIEW

**Comment by flatcap** [comment]
I've just rebased this onto master.

File: icommands.c [hunk 1]
```diff
@@ -180,5 +182,24 @@ static int icmd_set(struct Buffer *buf, struct Buffer *s, unsigned long data, st
...
+  }
+  struct Pager info = { 0 };
+  i = mutt_pager("set", tempfile, MUTT_PAGER_RETWINCH, &info);
+  if (!i)
+  {
+     mutt_error(_("Could not create temporary file"));
+     return 0;
+  }
+/*  mutt_enter_command(); */
   return 1;
```
Comment (yuzibo) [inline icommands.c]: First, my code look likes test :).
Now, i malloc a pager to show "set" variable.Next, how i should to get config items in config file.I know in 
`neomutt_parse_icommand` that calls `mutt_parse_rc_line`.
Thanks,

**Comment by flatcap** [comment]
All the config data lives in the `struct ConfigSet *Config;`
The functions in `config/set.h` will let you extract it in any way you like, but...

`dump_config()` in config/dump.c has all the code you need.
There are a bunch of flags in `config/dump.h` that control what gets displayed, such as `CS_DUMP_ONLY_CHANGED`.

Cut'n'paste what you need for now, then we can refactor things when you're done.

**Comment by flatcap** [comment]
@0x747972 
I've transferred your work onto this branch, tidied the code a little and rebased it onto `master`.

`:set`, `:set all` and `:version` work nicely, but currently they interfere with normal set commands,
e.g. `:set sidebar_visible` doesn't work.

**Comment by flatcap** [comment]
The 'summary pages' framework has been refactored and tidied.
The `:set [all]` and `:version` commands have been implemented in 908a11fa1 and 85dfaca6c.

**Pull request closed without merging**